docusaurus/docs/Android/02-client/06-guides/03-channel-list-updates.mdx

```diff
@@ -0,0 +1,124 @@
+
+You can fix that by providing two different event handlers.
+
+The first one is going to handle `public` channels updates:
+
+```kotlin
+class PublicChatEventHandler(
+    channels: StateFlow<Map<String, Channel>?>,
+    clientState: ClientState,
+) : DefaultChatEventHandler(channels, clientState) {
+
+    override fun handleChannelEvent(event: HasChannel, filter: FilterObject): EventHandlingResult {
+        // If the channel event matches "public" type, handle it
+        return if (event.channel.cid.startsWith("public")) {
+            super.handleChannelEvent(event, filter)
+        } else {
+            // Otherwise skip
+            EventHandlingResult.Skip
+        }
+    }
+
+    override fun handleCidEvent(
+        event: CidEvent,
+        filter: FilterObject,
+        cachedChannel: Channel?,
+    ): EventHandlingResult {
+        // If the cid event matches "public" type, handle it
+        return if (event.cid.startsWith("public")) {
+            super.handleCidEvent(event, filter, cachedChannel)
+        } else {
+            // Otherwise skip
+            EventHandlingResult.Skip
+        }
+    }
+}
+
+class PublicChatEventHandlerFactory : ChatEventHandlerFactory() {
+    override fun chatEventHandler(channels: StateFlow<Map<String, Channel>?>): ChatEventHandler {
+        return PublicChatEventHandler(channels, ChatClient.instance().clientState)
+    }
+}
+```
+
+The second one is for `private` channels updates:
+
+```kotlin
+class PrivateChatEventHandler(
+    channels: StateFlow<Map<String, Channel>?>,
+    clientState: ClientState,
+) : DefaultChatEventHandler(channels, clientState) {
+
+    override fun handleChannelEvent(event: HasChannel, filter: FilterObject): EventHandlingResult {
+        // If the channel event matches "private" type, handle it
+        return if (event.channel.cid.startsWith("private")) {
+            super.handleChannelEvent(event, filter)
+        } else {
+            // Otherwise skip
+            EventHandlingResult.Skip
+        }
+    }
+
+    override fun handleCidEvent(
+        event: CidEvent,
+        filter: FilterObject,
+        cachedChannel: Channel?,
+    ): EventHandlingResult {
+        // If the cid event matches "private" type, handle it
+        return if (event.cid.startsWith("private")) {
+            super.handleCidEvent(event, filter, cachedChannel)
+        } else {
+            // Otherwise skip
+            EventHandlingResult.Skip
+        }
+    }
+}
+
+class PrivateChatEventHandlerFactory : ChatEventHandlerFactory() {
+    override fun chatEventHandler(channels: StateFlow<Map<String, Channel>?>): ChatEventHandler {
+        return PrivateChatEventHandler(channels, ChatClient.instance().clientState)
+    }
+}
+```
+
+After that, we just need to apply the handlers to the proper `ChannelListViewModel`:
+
+```kotlin
+val factory = ChannelListViewModelFactory(chatEventHandlerFactory = chatEventHandlerFactory)
+```
```

Comment:
I would add some extra explanation at the bottom - it's usually always good to have a "where to go from here" or "what to do next" or "let's dive deeper" sections.

You can say something like:

"With this approach, you can make sure each list in your app only receives and handles the events it should be aware of. The `public` channel list will ignore `private` channel events and vice-versa.

You can use this for many different use cases, like handling different channel categories, groups, or custom types of metadata that describe your channels."

Reply:
good idea